### ISSUE TITLE
Problems with Reddit command.

### ISSUE THREAD

**Issue by RohanJnr** [python-discord/sir-lancebot#348]
Recently, the Reddit command PR (#346) was merged with a number of problems being overlooked. This issue is to outline the required fixes and any additional enhancements to the feature.
## Current Problems

- When navigating using reactions, the embed is edited twice in succession, first making the embed blank and then with the new content. This sometimes results in slow updating and the embed staying blank.
![here](https://cdn.discordapp.com/attachments/542272993192050698/676055841220067338/Screenshot_20200209-131812.jpg)

- The user-agent in the header dict is currently set to `Iceman` but in fact it should be set the app/client that is sending the request.
So it needs to be changed to : python3:python-discord/seasonalbot:1.0.0 (by /u/PythonDiscord)

- Currently, the API does not use an authenticated request. We should make sure to use an app token from Reddit the same as the Python Bot does also. This bot will need one created by an admin specific to Seasonalbot.
Use the PyDis Bot's Reddit cog implementation [here](https://github.com/python-discord/bot/blob/master/bot/cogs/reddit.py) as reference.

- The word `meant` is misspelt as `ment`([on line 60](https://github.com/python-discord/seasonalbot/blob/master/bot/seasons/evergreen/reddit.py#L60)).

**Comment by Suhail6inkling** [comment]
also - this is a minor error but

* “meant” is misspelt incorrectly ([shown on L60](https://github.com/python-discord/seasonalbot/blob/master/bot/seasons/evergreen/reddit.py#L60))

**Comment by scragly** [comment]
Definitely worth mentioning, thanks @Suhail6inkling.

**Comment by AG4lyf** [comment]
Bot doesn't check if the reddit have any post in it.
Such reddit can be `/r/makemyideas`
Bot fails silently 
I found this on private testing 
```
03/01/20 09:29:29 - bot.seasons.evergreen.error_handler ERROR: Unhandled command error: list index out of range
Traceback (most recent call last):
  File "/home/ag/.local/share/virtualenvs/seasonalbot-0Ba-uPuk/lib/python3.8/site-packages/discord/ext/commands/core.py", line 79, in wrapped
    ret = await coro(*args, **kwargs)
  File "/home/ag/PycharmProjects/bookmark-re-discord/seasonalbot/bot/seasons/evergreen/reddit.py", line 58, in get_reddit
    if posts[1]["data"]["over_18"] is True:
IndexError: list index out of range
```
It happened today 
https://discordapp.com/channels/267624335836053506/607247579608121354/683523350353674313

**Comment by quanta-kt** [comment]
Instead of checking if the first post is 18+ using `if posts[1]["data"]["over_18"] is True:` I'd rather `filter` out all the 18+ posts

**Comment by RohanJnr** [comment]
yes, that should also work @quanta-kt.

**Comment by RohanJnr** [comment]
I will be completing this along with #530

**Comment by ChrisLovering** [comment]
Fixed with #530
